Ticketing/Ticketing.UI/Controllers/VenuesController.cs

```diff
@@ -0,0 +1,97 @@
+﻿using Microsoft.AspNetCore.Mvc;
+using Microsoft.AspNetCore.OutputCaching;
+using Ticketing.BAL.Contracts;
+using Ticketing.BAL.Model;
+using log4net;
+
+namespace Ticketing.UI.Controllers
+{
+    /// <summary>
+    /// Venues API
+    /// </summary>
+    [Route("api/[controller]")]
+    [ApiController]
+    public class VenuesController : ControllerBase
+    {
+        private readonly IVenueService _venueService;
+        private readonly ILog _logger;
+
+        public VenuesController(IVenueService venueService, ILog logger)
+        {
+            _venueService = venueService;
+            _logger = logger;
+        }
+
+        /// <summary>
+        /// Get a list with all "<see cref="VenueReturnModel"/>" items.
+        /// <returns>Venues</returns>
+        /// <response code="200">Return collection of venues</response>
+        /// <response code="204">Return empty collection</response>
+        /// <response code="400">Bad request</response>        
+        /// </summary>
+        [HttpGet]
+        [OutputCache(PolicyName = "Expensive")]
+        public async Task<IActionResult> Get()
+        {
+            _logger.Info("VenuesControler  Start Get venues.");
+
+            var venues = await _venueService.GetVenuesAsync();
+
+            if (venues is null)
+            {
+                _logger.Error("VenuesControler Get venues Return BadRequest status.");
+                return BadRequest(string.Empty);
+            }
+            else if(!venues.Any())
+            {
+                _logger.Warn("VenuesControler Get venues Return empty result.");
```

Comment:
as this is expected -this in to wawrning. More looks like debug

Reply:
done